kong/db/dao/targets.lua

```diff
@@ -46,6 +47,15 @@ function _TARGETS:insert(entity, options)
     entity.target = formatted_target
   end
 
+  local workspace = workspaces.get_workspace_id()
+  local opts = { nulls = true, workspace = workspace }
+  for existent in self:each_for_upstream(entity.upstream, nil, opts) do
+    if existent.target == entity.target then
+      local err_t = self.errors:unique_violation({ target = existent.target })
```

Comment:
Does this need to be changed to update the existing `target` instead of returning an error as well? Will this be a breaking change?

Reply:
As we talked privately: the admin API first tries to update the entity before inserting it, the behavior for end-users does not change at all, only for DAO users. I think this does not need to be considered a breaking change.